plugins/hotkeyshortcuts.koplugin/main.lua

```diff
@@ -0,0 +1,581 @@
+        local remaining_keys = { "A", "S", "D", "F", "G", "H", "J", "K", "L", "Z", "X", "C", "V", "B", "N", "M" }
+        if Device.k3_alt_plus_key_kernel_translated then
+            -- Add the infamous top row keys, with kernel issues
+            for _, key in ipairs(top_row_keys) do
+                self.key_events["AltPlus" .. key] = {
+                    { Device.k3_alt_plus_key_kernel_translated[key] },
+                    event = "HotkeyAction",
+                    args = "alt_plus_" .. key:lower()
+                }
+            end
+        else
+            addKeyEvents(second_modifier, top_row_keys, "HotkeyAction", "alt_plus_")
+        end
+        addKeyEvents(second_modifier, remaining_keys, "HotkeyAction", "alt_plus_")
+    end -- if hasKeyboard()
+end -- registerKeyEvents()
+
+HotKeyShortcuts.onPhysicalKeyboardConnected = HotKeyShortcuts.registerKeyEvents
+
+function HotKeyShortcuts:shortcutTitleFunc(hotkey)
+    local title = hotkeyshortcuts_list[hotkey]
+    return T(_("%1: (%2)"), title, Dispatcher:menuTextFunc(self.hotkeyshortcuts[hotkey]))
+end
+
+function HotKeyShortcuts:genMenu(hotkey)
+    local sub_items = {}
+    if hotkeyshortcuts_list[hotkey] ~= nil then
+        table.insert(sub_items, {
+            text = T(_("%1 (default)"), Dispatcher:menuTextFunc(self.defaults[hotkey])),
+            keep_menu_open = true,
+            separator = true,
+            checked_func = function()
+                return util.tableEquals(self.hotkeyshortcuts[hotkey], self.defaults[hotkey])
+            end,
+            callback = function()
+                self.hotkeyshortcuts[hotkey] = util.tableDeepCopy(self.defaults[hotkey])
+                self.updated = true
+            end,
+        })
+    end
+    table.insert(sub_items, {
+        text = _("Pass through"),
+        keep_menu_open = true,
+        checked_func = function()
+            return self.hotkeyshortcuts[hotkey] == nil
+        end,
+        callback = function()
+            self.hotkeyshortcuts[hotkey] = nil
+            self.updated = true
+        end,
+    })
+    Dispatcher:addSubMenu(self, sub_items, self.hotkeyshortcuts, hotkey)
+    sub_items.max_per_page = nil -- restore default, settings in page 2
+    return sub_items
+end
+
+function HotKeyShortcuts:genSubItem(hotkey, separator, hold_callback)
+    local reader_only = {
+        -- these button combinations are used by different events in FM already, don't allow users to customise them.
+        modifier_plus_down = true,
+        modifier_plus_left_page_back = true,
+        modifier_plus_left_page_forward = true,
+        modifier_plus_right_page_back = true,
+        modifier_plus_right_page_forward = true,
+        modifier_plus_press = true,
+    }
+    if G_reader_settings:isTrue("press_key_does_hotkeyshortcuts") then
+        local fm_do_not_press = { press = true }
+        util.tableMerge(reader_only, fm_do_not_press)
```

Comment:
Why is it named "do_not" ?

Reply:
because you are not suppose to press it... i guess, i can't remember exactly why i named it that. Nevertheless it is never used again.